### PR TITLE
Refactor Group model

### PR DESCRIPTION
**Pull request by udan11**

### CODE REVIEW

File: app/assets/javascripts/discourse/models/group.js.es6 [hunk 1]
```diff
@@ -1,31 +1,31 @@
...
+import User from "discourse/models/user";
 
 const Group = RestModel.extend({
-  limit: 50,
-  offset: 0,
   user_count: 0,
+  limit: null,
+  offset: null,
+
+  request_count: 0,
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/models/group.js.es6]:
```suggestion
  requestCount: 0,
```

File: app/assets/javascripts/discourse/mixins/load-more.js.es6 [hunk 2]
```diff
@@ -21,6 +21,7 @@ export default Mixin.create(Scrolling, {
     const eyeline = new Eyeline(this.eyelineSelector + ":last");
     this.set("eyeline", eyeline);
     eyeline.on("sawBottom", () => this.send("loadMore"));
+    eyeline.update(); // Force update once to consider current position
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/mixins/load-more.js.es6]: This should be in another PR/commit as it's not really related to the group refactoring (remember we squash all the commits when we merge a PR)

**Comment by ZogStriP** [comment]
LGTM 👍

File: app/assets/javascripts/discourse/models/group.js.es6 (hunk 1 above)
Comment (udan11) [reply app/assets/javascripts/discourse/models/group.js.es6]: This is similar to `user_count` above. We are building the model from the exact response we get from the server.

**Comment by ZogStriP** [comment]
Having some tests would be great here as well ;)
